cmd/soci/commands/push.go

```diff
@@ -109,11 +112,28 @@ if they are available in the snapshotter's local content store.
 				return indexDescriptors[i].CreatedAt.Before(indexDescriptors[j].CreatedAt)
 			})
 
-			username := cliContext.String("user")
+			refspec, err := reference.Parse(ref)
```

Comment:
Could we move this outside of the loop?

Reply:
re: Actually I think we can move the getting credentials outside the loop entirely.

Reply:
This was actually pre-existing code that I moved, but yes, this and half the rest of the loop can be moved outside the loop. I'll do that.